### ISSUE TITLE
XSimGCL critical failure on other datasets

### ISSUE THREAD

**Issue by KlayZhang05** [Coder-Yu/SELFRec#78]
I encountered critical failures when I tried to test XSimGCL on the yelp2018 train.txt and other external datasets. All the metrics show 0.0, which means the model can't hit even one correct recommendation. ​​
Observing the train.txt and test.txt of yelp2018, it is found that the data range of the items interacted by user 0 in the training set is **0-20**, while the data range of the items interacted by user 0 in the test set is **several hundred larger**, and the prediction of XSimGCL can hit such a test set?
Continuing to observe the raw output and prediction set, it is found that the recommendations made by the model are all cold-start products (I understand that the user in the training set did not interact). This resulted in a steady 0 for all metrics, both when I was training on my own dataset and testing on the YELP training set.

Since the the training set and the test set of yelp2018 are too different, how do you split the dataset? How can I apply my dataset on XSimGCL?

Appreciate a lot.

PS: I attached my dataset so you can see the difference

[test.txt](https://github.com/user-attachments/files/19460267/test.txt)
[train.txt](https://github.com/user-attachments/files/19460266/train.txt)

.

**Comment by Coder-Yu** [comment]
If you used the built-in yelp2018 dataset, there should be no errors.
Did you use your own dataset? If you do, please check your split code.

**Comment by KlayZhang05** [comment]
> If you used the built-in yelp2018 dataset, there should be no errors. Did you use your own dataset? If you do, please check your split code.
大佬你好，我用了我自己的数据集，由于我不知道yelp2018是怎么划分的，因此我沿用了我想要对比的模型的划分方法，即按时间阶段划分，以最后一个阶段为测试集（我的数据集本来是sequential data，我把时间戳去掉了）。我发现XSimGCL只能在yelp2018的测试集上表现正常，不论是用yelp2018的训练集测试，还是在我自己的数据集上测试，HR、Recall等都是0.0.
这表明XSimGCL预测结果全是冷启动样本？

**Comment by Coder-Yu** [comment]
这表明你的划分有问题。
其次训练集不是用来测试的，我的库里评估算法的时候训练集里面的项目得分会重置为0，如果不这样，这些项目显而易见的会高于测试集项目。你用训练集测试，自然指标都是0。
建议你自己再思考一下整个评估流程，检查一下你的数据集划分方法。
